### PR TITLE
IcebergTableDataObject: throw NoDataToProcessWarning if no snapshot is created (#878)

### PR DESCRIPTION
**Pull request by zzeekk**
see #878

### CODE REVIEW

File: sdl-deltalake/src/test/scala/io/smartdatalake/workflow/dataobject/DeltaLakeTableDataObjectTest.scala [hunk 1]
```diff
@@ -19,17 +19,18 @@
...
 
 import io.smartdatalake.config.InstanceRegistry
-import io.smartdatalake.definitions.{ColumnStatsType, SDLSaveMode, SaveModeMergeOptions, TableStatsType}
+import io.smartdatalake.definitions.{ColumnStatsType, Environment, SDLSaveMode, SaveModeMergeOptions, TableStatsType}
 import io.smartdatalake.testutils.custom.TestCustomDfCreator
 import io.smartdatalake.testutils.{MockDataObject, TestUtil}
 import io.smartdatalake.util.hdfs.{HdfsUtil, PartitionValues}
 import io.smartdatalake.util.spark.DataFrameUtil.DfSDL
-import io.smartdatalake.workflow.action.CopyAction
+import io.smartdatalake.workflow.action.{CopyAction, NoDataToProcessWarning}
```
Comment (Geheiner) [inline sdl-deltalake/src/test/scala/io/smartdatalake/workflow/dataobject/DeltaLakeTableDataObjectTest.scala]: as there are no other changes in this file I would be surpirzed if these imports are necessary. Can you run optimize imports for this File?

File: sdl-iceberg/src/main/scala/io/smartdatalake/workflow/dataobject/IcebergTableDataObject.scala [hunk 2]
```diff
@@ -318,6 +319,8 @@ case class IcebergTableDataObject(override val id: DataObjectId,
 
     val finalSaveMode = saveModeOptions.map(_.saveMode).getOrElse(saveMode)
     val saveModeTargetDf = saveModeOptions.map(_.convertToTargetSchema(dfPrepared)).getOrElse(dfPrepared)
+    // remember previous snapshot timestamp
+    val previousSnapshotId = if (isTableExisting) Some(getIcebergTable.currentSnapshot().snapshotId()) else None
```
Comment (Geheiner) [inline sdl-iceberg/src/main/scala/io/smartdatalake/workflow/dataobject/IcebergTableDataObject.scala]: can you add the type of this val? looks like a String but I'm not sure

File: sdl-iceberg/src/main/scala/io/smartdatalake/workflow/dataobject/IcebergTableDataObject.scala [hunk 3]
```diff
@@ -364,8 +367,14 @@ case class IcebergTableDataObject(override val id: DataObjectId,
     })
 
     // get iceberg snapshot summary / stats
-    val summary = getIcebergTable.currentSnapshot().summary().asScala
-    assert(summary("spark.app.id") == session.sparkContext.applicationId, s"($id) current iceberg snapshot is not written by this spark application (spark.app.id should be ${summary("spark.app.id")}. Is there someone else writing to this table?!")
+    val currentSnapshot = getIcebergTable.currentSnapshot()
+    if (logger.isDebugEnabled) logger.debug(s"snapshot after write: ${currentSnapshot.toString}")
+    val summary = currentSnapshot.summary().asScala
+    if (previousSnapshotId.contains(currentSnapshot.snapshotId())) {
```
Comment (Geheiner) [inline sdl-iceberg/src/main/scala/io/smartdatalake/workflow/dataobject/IcebergTableDataObject.scala:373]: why compare 2 strings with contains? Is one really a subset of the other?

**Review by Geheiner — APPROVED** [review]
Minor cleanups suggested, but OK

File: sdl-iceberg/src/main/scala/io/smartdatalake/workflow/dataobject/IcebergTableDataObject.scala (hunk 3 above)
Comment (zzeekk) [reply sdl-iceberg/src/main/scala/io/smartdatalake/workflow/dataobject/IcebergTableDataObject.scala:373]: previousSnapshotId is an Option[Long] - hard to see if not in an IDE, i agree.

File: sdl-iceberg/src/main/scala/io/smartdatalake/workflow/dataobject/IcebergTableDataObject.scala (hunk 2 above)
Comment (zzeekk) [reply sdl-iceberg/src/main/scala/io/smartdatalake/workflow/dataobject/IcebergTableDataObject.scala]: Ok, i'll add it. It's Option[Long]

**Pull request merged**